From e27c9e67efb9607d4330438deb458d4624d5104c Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 15 Dec 2022 11:33:02 +0000
Subject: [PATCH 08/13] Update paths for presigned shim

The presigned shim from Fedora is hardcoded to boot grubx64.efi,
masquerade systemd-boot as such.
---
 src/bootloaders/shim-systemd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/bootloaders/shim-systemd.c b/src/bootloaders/shim-systemd.c
index 43fa417..3485722 100644
--- a/src/bootloaders/shim-systemd.c
+++ b/src/bootloaders/shim-systemd.c
@@ -37,7 +37,7 @@
  *
  *          KERNEL_NAMESPACE/
  *              bootloaderx64.efi       <-- shim
- *              loaderx64.efi           <-- systemd-boot bootloader
+ *              grubx64.efi             <-- systemd-boot bootloader (The pre-signed shim is hardcoded to boot grubx64.efi)
  *              mmx64.efi               <-- MOK manager
  *              fbx64.efi               <-- fallback bootloader
  *
@@ -124,7 +124,7 @@ __cbm_export__ const BootLoader
 
 /* these path components can be used as-is, no need to probe */
 #define SHIM_DST "bootloader" EFI_SUFFIX
-#define SYSTEMD_DST "loader" EFI_SUFFIX
+#define SYSTEMD_DST "grub" EFI_SUFFIX
 #define SYSTEMD_CONFIG_DIR "loader"
 #define SYSTEMD_ENTRIES_DIR "entries"
 
-- 
2.40.1

